### PR TITLE
Jormun: Refacto street_network_configurations

### PR DESCRIPTION
**Pull request by AurelienLP**
We'll want to manage backends in db via Tyr and Tyr configurator. To do so, we want to have one class which will update the configurations every time we need it (when we call a backend).

Like `EquipmentProviderManager`, I have created a class `StreetNetworkBackendManager`.

For now, I only want to keep the retro compatibility with the actual json system.

This PR is only a refacto and shouldn't introduce any regression.

It is reviewable commit by commit !

### CODE REVIEW

File: source/jormungandr/jormungandr/instance.py [hunk 1]
```diff
@@ -162,10 +133,9 @@ def __init__(
         self.georef = georef.Kraken(self)
         self.planner = planner.Kraken(self)
 
-        street_network_configurations = _set_default_street_network_config(street_network_configurations)
-        self.street_network_services = street_network.StreetNetwork.get_street_network_services(
-            self, street_network_configurations
-        )
+        self.streetnetwork_backend_manager = StreetNetworkBackendManager(street_network_configurations)
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/instance.py]: I would prefer having a manager shared between instances to be able to use the same instance of the backend class so circuit breaker, caches and others things can be shared.

**Comment by AurelienLP** [comment]
Jenkins, test this please

**Comment by AurelienLP** [comment]
jenkins, test this

**Comment by AurelienLP** [comment]
Jenkins, test this please

File: source/jormungandr/jormungandr/instance.py [hunk 2]
```diff
@@ -49,7 +49,7 @@
 from jormungandr.exceptions import DeadSocketException
 from navitiacommon import models
 from importlib import import_module
-from jormungandr import cache, memory_cache, app, global_autocomplete
+from jormungandr import cache, memory_cache, app, global_autocomplete, streetnetwork_backend_manager
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/instance.py]: if possible inject it as a dependency

Comment (AurelienLP) [reply source/jormungandr/jormungandr/instance.py]: Done

**Review by kinnou02 — APPROVED** [review]

File: source/jormungandr/jormungandr/instance.py [hunk 3]
```diff
@@ -200,6 +169,9 @@ def __init__(
         # Init equipment providers from config
         self.equipment_provider_manager.init_providers(instance_equipment_providers)
 
+        self._get_street_network = streetnetwork_backend_manager.get_street_network
```
Comment (xlqian) [inline source/jormungandr/jormungandr/instance.py]: :smiling_imp:

File: source/jormungandr/jormungandr/street_network/streetnetwork_backend_manager.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            instance_configuration.append(copy.deepcopy(config))
+
+        return instance_configuration
+
+    def _create_street_network_backends(self, instance, instance_configuration):
+        for config in instance_configuration:
+            # Set default arguments
+            if 'args' not in config:
+                config['args'] = {}
+            if 'service_url' not in config['args']:
```
Comment (xlqian) [inline source/jormungandr/jormungandr/street_network/streetnetwork_backend_manager.py]: config['args'].setdefault('service_url', None)
config['args'].setdefault('instance', instance)
config['args'].setdefault('modes', config.get('modes', []))

File: source/jormungandr/jormungandr/instance.py (hunk 3 above)
Comment (AurelienLP) [reply source/jormungandr/jormungandr/instance.py]: Does it make the code harder to read? I should maybe find an other way...

**Review by xlqian — APPROVED** [review]

**Comment by AurelienLP** [comment]
Jenkins, test this please

**Pull request merged**